agentcfg/cache.go

```diff
@@ -0,0 +1,81 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package agentcfg
+
+import (
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+const (
+	cleanupInterval time.Duration = 60 * time.Second
+)
+
+type cache struct {
+	logger  *logp.Logger
+	exp     time.Duration
+	gocache *gocache.Cache
+}
+
+func newCache(logger *logp.Logger, exp time.Duration) *cache {
+	if logger == nil {
```

Comment:
this seems an unnecessary complication, rather pass the logger in `cache_test/newCacheSetup`

Reply:
I don't understand, passing in the logger here has nothing to do with any testing logging. I just pass the logger from the fetcher to the cache.

Reply:
sure, i just meant the `if nil` check

Reply:
The `logger` is a pointer, so it could be `nil`. I know, calling the function it is under our control, so one can argue it is not necessary as we always pass a logger in.. but I don't think it hurts and I rather avoid panics than being lazy with nil checks.